### PR TITLE
Add `JENKINS_UC_DOWNLOAD`, allows overriding plugin download URL

### PR DESCRIPTION
**Pull request by oganga**
Possible fix for #191

### CODE REVIEW

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java [hunk 1]
```diff
@@ -909,8 +909,10 @@ public String getPluginDownloadUrl(Plugin plugin) {
             pluginVersion = "latest";
         }
 
-        if (!StringUtils.isEmpty(pluginUrl)) {
-            logVerbose(String.format("Will use url: %s to download %s plugin", pluginUrl, plugin.getName()));
+        String jenkinsUcDownload =  System.getenv("JENKINS_UC_DOWNLOAD");
+        if (StringUtils.isNotEmpty(jenkinsUcDownload)){
+            urlString = appendPathOntoUrl(jenkinsUcDownload, "/download/plugins", pluginName, pluginVersion, pluginName + ".hpi");
+        }else if (!StringUtils.isEmpty(pluginUrl) ) {
```
Comment (jetersen) [inline plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java:915]:
```suggestion
        String jenkinsUcDownload =  System.getenv("JENKINS_UC_DOWNLOAD");
        if (StringUtils.isNotEmpty(jenkinsUcDownload)) {
            urlString = appendPathOntoUrl(jenkinsUcDownload, "/download/plugins", pluginName, pluginVersion, pluginName + ".hpi");
        } else if (StringUtils.isNotEmpty(pluginUrl)) {
```

**Review by jetersen — COMMENTED** [review]
While your at it, use `isNotEmpty` and fix formatting

**Comment by jetersen** [comment]
Så fortjener du lige en. Skide godt, Thorkild! 😆

**Review by timja — COMMENTED** [review]
needs docs

**Comment by oganga** [comment]
@timja Yes, regarding the doc,, I am still a bit unsure if env var is the correct way of doing it or you(the project) would prefer a command line flag?
There is a bt more work in doing the command line flag.

File: README.md [hunk 2]
```diff
@@ -51,6 +51,9 @@ Jenkins in a broken state.
 * `CACHE_DIR`: used to configure the directory where the plugins update center cache is located. By default it will be in `~/.cache/jenkins-plugin-management-cli`,
 if the user doesn't have a home directory when it will go to: `$(pwd)/.cache/jenkins-plugin-management-cli`.
 
+* `JENKINS_UC_DOWNLOAD`: used to configure the URL from where plugins will be downloaded from. Often used to cache or as proxy the Jenkins plugin download site.
```
Comment (timja) [inline README.md:54]:
```suggestion
* `JENKINS_UC_DOWNLOAD`: used to configure the URL from where plugins will be downloaded from. Often used to cache or to proxy the Jenkins plugin download site. 
```

**Review by timja — APPROVED** [review]

**Comment by oganga** [comment]
What is the time frame to get this in an release and into the Jenkins Docker image?
Just to get an idea of if we should wait or do some in house manual stuff to fix our issue.

**Comment by timja** [comment]
released, PR here for adding to the docker image:
https://github.com/jenkinsci/docker/pull/992

you can easily override it by redownloading the jar with the updated version in the meantime

**Comment by oganga** [comment]
Cool!!
Thanks for the quick response!

**Pull request merged**